File: 121 Best Time to Buy and Sell Stock/Java/Solution.java
public class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        int idx = 0;
        for (int i = 0; i < prices.length; i++) {
            int smallestPrice = prices[idx];
            int currentPrice = prices[i];
            if (currentPrice < smallestPrice) {
                idx = i;
            }
            else {
                max = Math.max(max, currentPrice - smallestPrice);
            }
        }
        return max;
    }
}

File: 121 Best Time to Buy and Sell Stock/C/solution.c
#include "math.h"

int maxProfit(int* prices, int pricesSize) {
    int m = 0;
    int idx = 0;
    for (int i = 0; i < pricesSize; i++) {
        int smallestPrice = prices[idx];
        int currentPrice = prices[i];
        if (currentPrice < smallestPrice) {
            idx = i;
        }
        else {
            m = max(m, currentPrice - smallestPrice);
        }
    }
    return m;
}

